src/CryptoMonitor.php

```diff
@@ -1,28 +1,54 @@
 <?php
-declare(strict_types=1);
 
+declare(strict_types=1);
 namespace PayCrypto;
 
+use PayCrypto\PayCrypto;
 use ScriptFUSION\Porter\Connector\Connector;
 use ScriptFUSION\Porter\Provider\Provider;
 
 class CryptoMonitor implements Provider
 {
-    public const EXCHANGE_API_ENDPOINT = 'https://rest.coinapi.io/';
-
     private $connector;
 
+    private static $providers = [
+        'coinapi' => [
+            'url' => 'https://rest.coinapi.io/v1/',
+            'specific' => 'exchangerate/%s/%s/',
+            'all' => 'exchangerate/%s/',
+            'assets' => 'assets/'
+        ]
+    ];
+
     public function __construct(Connector $connector)
     {
         $this->connector = $connector;
     }
 
-    public static function buildExchangeApiUrl(string $url): string
+    public static function buildExchangeApiUrl(PayCrypto $config) : string
     {
-        return self::EXCHANGE_API_ENDPOINT . $url;
+        $url = self::$providers[$config->provider]['url'];
+
+        $params = self::$providers[$config->provider][$config->type];
+
+        switch ($config->type) {
```

Comment:
How about enhancing this `type` condition?

I think using the `switch...case` is not the proper way to do.

If we have the many types to do the different action, the `swittch` condition will be very long.